signing/signing.go

```diff
@@ -0,0 +1,106 @@
+		FulcioURL:    sigOpts.DefaultFulcioURL,     // Signing certificate provider.
+		RekorURL:     sigOpts.DefaultRekorURL,      // Transparency log.
+		OIDCIssuer:   sigOpts.DefaultOIDCIssuerURL, // OIDC provider to get ID token to auth for Fulcio.
+		OIDCClientID: "sigstore",
+	}
+	regOpts := sigOpts.RegistryOptions{} // Not necessary so we leave blank.
+
+	// This command will use the provided OIDCIssuer to authenticate into Fulcio, which will generate the
+	// signing certificate on the scorecard result. This attestation is then uploaded to the Rekor transparency log.
+	// The output bytes (signature) and certificate are discarded since verification can be done with just the payload.
+	if _, err := sign.SignBlobCmd(rootOpts, keyOpts, regOpts, scorecardResultsFile, true, "", ""); err != nil {
+		return fmt.Errorf("error signing payload: %w", err)
+	}
+
+	return nil
+}
+
+// Changes output settings to json and runs scorecard again.
+// TODO: run scorecard only once and generate multiple formats together.
+func GetJsonScorecardResults() ([]byte, error) {
+	defer os.Setenv(options.EnvInputResultsFile, os.Getenv(options.EnvInputResultsFile))
+	defer os.Setenv(options.EnvInputResultsFormat, os.Getenv(options.EnvInputResultsFormat))
+	os.Setenv(options.EnvInputResultsFile, "results.json")
+	os.Setenv(options.EnvInputResultsFormat, "json")
+
+	actionJson, err := entrypoint.New()
+
+	if err != nil {
+		return nil, fmt.Errorf("creating scorecard entrypoint: %v", err)
+	}
+	if err := actionJson.Execute(); err != nil {
+		return nil, fmt.Errorf("error during command execution: %v", err)
+	}
+
+	// Get json output data from file.
+	jsonPayload, err := ioutil.ReadFile(os.Getenv(options.EnvInputResultsFile))
+	if err != nil {
+		return nil, fmt.Errorf("reading scorecard json results from file: %v", err)
+	}
+
+	return jsonPayload, nil
+}
+
+// Calls scorecard-api to process & upload signed scorecard results.
+func ProcessSignature(sarifPayload []byte, jsonPayload []byte, repoName string, repoRef string) error {
+
+	// Prepare HTTP request body for scorecard-webapp-api call.
+	resultsPayload := struct {
+		SarifOutput string
+		JsonOutput  string
+	}{
+		SarifOutput: string(sarifPayload),
+		JsonOutput:  string(jsonPayload),
+	}
+
+	payloadBytes, err := json.Marshal(resultsPayload)
+	if err != nil {
+		return fmt.Errorf("reading scorecard json results from file: %v", err)
+	}
+
+	// Call scorecard-webapp-api to process and upload signature.
+	url := "https://api.securityscorecards.dev/verify"
+	req, _ := http.NewRequest("POST", url, bytes.NewBuffer(payloadBytes))
+	req.Header.Set("X-Repository", repoName)
+	req.Header.Set("X-Branch", repoRef)
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("executing scorecard-api call: %v", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		bodyBytes, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("http response error: %v", string(bodyBytes))
```

Comment:
might be worth adding the `resp.Status` and `resp.StatusCode` in the error string -- in case there's not body in the not OK HTTP response, it will be hard to debug whether this was a 5xx or 4xx or other type of error.